### ISSUE TITLE
only use mosquitto local_cleansession if mosquitto version supports it

### ISSUE THREAD

**Issue by reubenmiller** [thin-edge/thin-edge.io#2184]
**Is your feature improvement request related to a problem? Please describe.**

After merging https://github.com/thin-edge/thin-edge.io/pull/2171, it was later found out that not all mosquitto versions support the `local_cleansession` bridge configuration leading to the mosquitto service failing to start (seen on mosquitto version `1.5.7`).

mosquitto added the bridge setting `local_cleansession` in https://github.com/eclipse/mosquitto/pull/1583, however it is unclear from which mosquitto versions the feature is available from. The error manifests with the following error:

```sh
$ /usr/sbin/mosquitto -c /etc/mosquitto/mosquitto.conf
Loading config file /etc/tedge/mosquitto-conf/c8y-bridge.conf
Error: Unknown configuration variable "local_cleansession".
Error found at /etc/tedge/mosquitto-conf/c8y-bridge.conf:12.
Error found at /etc/mosquitto/mosquitto.conf:13.
Error: Unable to open configuration file.
```

**Describe the solution you'd like**

The mosquitto version needs to be auto detected and only use the `local_cleansession` property if the mosquitto version is >= 2.x. However mosquitto does not have a very reliable way to work out which version is running (except for running `mosquitto --help` and parsing the output).

**Describe alternatives you've considered**

**Additional context**

**Comment by jhoo-sag** [comment]
Is there a workaround for this issue? Currently facing this with Thin Edge v0.12 and mosquitto v1.6.8 on a partner gateway and I can't figure out how to permanently overwrite the option.

**Comment by didier-wenzek** [comment]
> Is there a workaround for this issue?

Unfortunately, I see no workaround.

> Currently facing this with Thin Edge v0.12 and mosquitto v1.6.8 on a partner gateway and I can't figure out how to permanently overwrite the option.

We can consider to fix that for 0.13

**Comment by reubenmiller** [comment]
I just did some testing of my own and it seems that even the latest mosquitto 1.x version (1.6.15) does not support the `local_cleansession` option...this makes support older distributions such as Debian Buster problematic as [1.5.7](https://packages.debian.org/buster/mosquitto) is the current version offered there.

I think the only way forward is to have the setting configurable from the tedge.toml...that way at least users can opt out of the setting so they can get everything to work.

**Comment by reubenmiller** [comment]
> Is there a workaround for this issue? Currently facing this with Thin Edge v0.12 and mosquitto v1.6.8 on a partner gateway and I can't figure out how to permanently overwrite the option.

I'm assuming you are using a version number like `0.12.0~xxx` right? As the latest official `0.12.0` version does not have this feature yet.

**Comment by didier-wenzek** [comment]
The mosquitto `local_cleansession` bridge property is a key setting to avoid loosing outgoing messages. However, with mosquitto < 2.0 this property cannot be set independently of the bridge `cleansession` property which has to be to true for c8y.

Hence, we decided to add a new `tedge config` setting .

`c8y.include_localcleansession` can be set to either: `true | false | auto`
- `c8y.include_localcleansession = false` is the default.  No `local_cleansession` property is added the to bridge configuration file which can then be used with any version of mosquiito.
- `c8y.include_localcleansession = auto` is the recommended setting. With this setting, the `tedge connect c8y` command inserts a  `local_cleansession` setting but only if the version of mosquitto is above 2.0.
- `c8y.include_localcleansession = true` can be use to force the appropriate setting of `local_cleansession` when `tedge` fails to detect that mosquitto is running a version above 2.0.

**Comment by reubenmiller** [comment]
QA has thoroughly checked the feature and here are the results:

- [x] Test for ticket exists in the test suite.
- [x] QA has tested the bug and could not reproduce it anymore.

**Comment by reubenmiller** [comment]
@jhoo-sag this fix now available on the main branch and it will be included in the 0.13.0 release that we plan to do on the 3rd Nov.
